File: 45.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        n = len(nums)
        res = 0
        segend = 0
        maxreach = 0
        for i in range(n - 1):
            maxreach = max(maxreach, i + nums[i])
            if i == segend:
                res += 1
                segend = maxreach
        return res

File: sol/45/45.cpp
struct Solution {
    inline int jump(vector<int>& nums) {
        int n = nums.size(), res = 0, segend = 0, maxreach = 0;
        for (int i = 0; i < n - 1; ++i){
            maxreach = max(maxreach, i + nums[i]);
            if (i == segend){
                ++res;
                segend = maxreach;
            }
        }
        return res;
    }
};
